pkgs/servers/search/elasticsearch/7.x.nix

```diff
@@ -46,38 +41,43 @@ stdenv.mkDerivation (rec {
       "ES_CLASSPATH=\"\$ES_CLASSPATH:$out/\$additional_classpath_directory/*\""
   '';
 
-  buildInputs = [ makeWrapper jre_headless util-linux ]
-             ++ optional enableUnfree zlib;
+  nativeBuildInputs = [ autoPatchelfHook ];
+
+  buildInputs = [ makeWrapper jre_headless util-linux zlib ];
+
+  runtimeDependencies = [ zlib ];
+
+  dontPatchELF = true;
 
   installPhase = ''
     mkdir -p $out
     cp -R bin config lib modules plugins $out
 
     chmod +x $out/bin/*
 
+    substituteInPlace $out/bin/elasticsearch \
+      --replace 'bin/elasticsearch-keystore' "$out/bin/elasticsearch-keystore"
+
     wrapProgram $out/bin/elasticsearch \
       --prefix PATH : "${makeBinPath [ util-linux coreutils gnugrep ]}" \
       --set JAVA_HOME "${jre_headless}"
 
     wrapProgram $out/bin/elasticsearch-plugin --set JAVA_HOME "${jre_headless}"
   '';
 
-  passthru = { inherit enableUnfree; };
+  postFixup = lib.optionalString stdenv.hostPlatform.isLinux ''
+    for exe in $(find $out/modules/x-pack-ml/platform/${plat}-${arch}/bin -executable -type f); do
+      echo "patching $exe..."
```

Comment:
Doesn't patchelf echo what it is doing?

Reply:
I don't see what this `postFixup` hook is doing that `autoPatchelfHook` doesn't, so I'm now removing it.